frontend/src/views/Blacksmith.vue

```diff
@@ -1076,25 +1079,45 @@ export default Vue.extend({
       }
     },
 
+    selectAllForBurn(){
+      if(this.burnWeaponIds.length > 0){
+        this.burnWeaponIds.forEach(id => {
+          this.ctr += 1;
+          const weaponDetails = this.ownWeapons.find(y => {
+            if(y && +y.id === +id){
+              return y;
+            }
+          });
+          if(weaponDetails){
+            this.computeDust('sub',(weaponDetails.stars + 1), weaponDetails.lowStarBurnPoints, weaponDetails.fourStarBurnPoints,
+              weaponDetails.fiveStarBurnPoints);
+          }
+        });
+        this.burnWeaponIds = [];
+        return;
+      }
+      this.ownedWeaponIds.map(id => this.burnWeaponIds.push(id));
+      this.burnWeaponIds.forEach(id => {
+        this.ctr += 1;
+        const weaponDetails = this.ownWeapons.find(y => +y.id === +id);
+        this.computeDust('add',(weaponDetails.stars + 1), weaponDetails.lowStarBurnPoints, weaponDetails.fourStarBurnPoints, weaponDetails.fiveStarBurnPoints);
+      });
+    },
```

Comment:
This select all doesn't account for filters or pagination. Select all should only select items visible at that time so should be done on the filtered list not the entire weapon list.